src/RewardsStreamerMP.sol

```diff
@@ -73,7 +96,91 @@ contract RewardsStreamerMP is UUPSUpgradeable, IStakeManager, TrustedCodehashAcc
         _checkOwner();
     }
 
-    function stake(uint256 amount, uint256 lockPeriod) external onlyTrustedCodehash onlyNotEmergencyMode nonReentrant {
+    /**
+     * @notice Check if a vault is registered
+     * @param vault The address of the vault to check
+     * @return true if the vault is registered, false otherwise
+     */
+    function isVaultRegistered(address vault) public view returns (bool) {
+        return stakeVaultRegistry.vaultOwners(vault) != address(0);
+    }
+
+    /**
+     * @notice Registers a vault with its owner. Called by the vault itself during initialization.
+     * @dev Only callable by contracts with trusted codehash
+     */
+    function registerVault() external onlyTrustedCodehash {
+        address vault = msg.sender;
+        address owner = IStakeVault(vault).owner();
+        stakeVaultRegistry.register(owner, vault);
+    }
+
+    /**
+     * @notice Get all vaults owned by an address
+     */
+    function getUserVaults(address owner) external view returns (address[] memory) {
+        return stakeVaultRegistry.vaultsOf(owner);
+    }
+
+    /**
+     * @notice Get the total multiplier points for a user
+     * @dev Iterates over all vaults owned by the user and sums the multiplier points
+     * @param user The address of the user
+     * @return The total multiplier points for the user
+     */
+    function getUserTotalMP(address user) external view returns (uint256) {
+        address[] memory userVaults = stakeVaultRegistry.vaultsOf(user);
+        uint256 userTotalMP = 0;
+
+        for (uint256 i = 0; i < userVaults.length; i++) {
+            Account storage account = accounts[userVaults[i]];
+            userTotalMP += account.accountMP + _getAccruedMP(account);
+        }
+        return userTotalMP;
+    }
+
+    /**
+     * @notice Get the total maximum multiplier points for a user
+     * @dev Iterates over all vaults owned by the user and sums the maximum multiplier points
+     * @param user The address of the user
+     * @return The total maximum multiplier points for the user
+     */
+    function getUserTotalMaxMP(address user) external view returns (uint256) {
```

Comment:
This should be in registry as well

Reply:
Hm.. not sure about that...
So far we said that the staking protocol ultimately is one of the "XP providers" for the XP token contract.
Therefore, the stakemanager has to expose these. StakeManager is also where rewards are distributed. 

I think it makes sense to keep the registry solely as registry for "which account owns which vaults".

Reply:
There's a tradeoff:

Do users/apps have to worry about that there's a registry under the hood or not.

With the current implementation, calls always go through the stake manager and it knows whether or not it has to talk to a registry that users don't have to know about.

It also ensures that StakeVaults that interact with the system are whitelisted.

The path you're suggesting means:

- Registry has to whitelist as well, because we need to be sure only legit stakevault interact with the system
- Apps have to talk to one more contract to figure things out

I'm personally in favor of the former, but if there's majority consensus I'm happy to make the changes.

@gravityblast maybe you can leave your opinion here as well?

Reply:
I would keep the functions in the staking manager and keep the registry only as a registry of vaults